lib/msf/core/db_manager/vuln.rb

```diff
@@ -179,9 +179,11 @@ def report_vuln(opts)
 
     # Try to match based on vuln_details records
     if not vuln and opts[:details_match]
-      vuln = find_vuln_by_details(opts[:details_match], host, service)
-      if vuln and service and not vuln.service
-        vuln.service = service
+      if service
+        vuln = find_vuln_by_details(opts[:details_match], host, service)
+        if vuln and not vuln.service
+          vuln.service = service
+        end
```

Comment:
❔  It's late here, so I might be missing something, but it seems like we'd no longer be executing the end logic of `find_vuln_by_details` with this code change?

https://github.com/rapid7/metasploit-framework/blob/b4c383506270179298393a80cda4ccfff26e1bd1/lib/msf/core/db_manager/vuln.rb#L33-L44

From what I understand `find_vuln_by_details` is already tolerant of `service` being `nil` - and `find_vuln_by_details` seems to be only called by this file, it looks like it's now dead code that can be deleted or something needs tweaked still? 🤔

Reply:
Took another look and found this could be re-adjusted to ensure a vuln can still be reported with no service attached.